tests/rados/test_omap_entries.py

```diff
@@ -25,60 +25,92 @@ def run(ceph_cluster, **kw):
+                # calculate objects to be written with omaps and begin omap creation process
+                omap_obj_num = (
+                    omap_target_configs[omap_config]["obj_end"]
+                    - omap_target_configs[omap_config]["obj_start"]
+                )
+                log.debug(
+                    "Created the pool. beginning to create large number of omap entries on the pool"
+                )
+                if not pool_obj.fill_omap_entries(
+                    pool_name=entry["pool_name"], **omap_target_configs[omap_config]
+                ):
+                    log.error(
+                        f"Omap entries not generated on pool {entry['pool_name']}"
+                    )
+                    raise Exception(
+                        f"Omap entries not generated on pool {entry['pool_name']}"
+                    )
 
-            # Waiting for cluster to get clean state after OSD stopped
-            if not wait_for_clean_pg_sets(rados_obj):
-                log.error("PG's in cluster are not active + Clean state.. ")
-                return 1
-            rados_obj.change_osd_state(action="restart", target=osd_id)
-            log.debug(
-                f"Cluster reached clean state after osd {osd_id} stop and restart"
-            )
+                assert pool_obj.check_large_omap_warning(
+                    pool=entry["pool_name"],
+                    obj_num=omap_obj_num,
+                    check=omap_target_configs[omap_config]["large_warn"],
+                )
+
+            # Fetching the current acting set for the pool
+            acting_set = rados_obj.get_pg_acting_set(pool_name=entry["pool_name"])
+            rados_obj.change_recover_threads(config={}, action="set")
+            log.debug(f"Proceeding to restart OSDs from the acting set {acting_set}")
+            for osd_id in acting_set:
+                rados_obj.change_osd_state(action="stop", target=osd_id)
+                # sleeping for 5 seconds for re-balancing to begin
+                time.sleep(5)
 
-        rados_obj.change_recover_threads(config={}, action="rm")
-        # deleting the pool created after the test
-        rados_obj.detete_pool(pool=entry["pool_name"])
+                # Waiting for cluster to get clean state after OSD stopped
+                if not wait_for_clean_pg_sets(rados_obj):
+                    log.error("PG's in cluster are not active + Clean state.. ")
+                    raise Exception("PG's in cluster are not active + Clean state.. ")
+                rados_obj.change_osd_state(action="restart", target=osd_id)
+                log.debug(
+                    f"Cluster reached clean state after osd {osd_id} stop and restart"
+                )
+        except Exception as e:
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            return 1
+        finally:
+            if entry.get("pool_type", "replicated") != "erasure":
```

Comment:
Is there any particular reason that we are coding steps only for replicated pools?

I agree that ec pool has been commented off as of now, but once it is resolved, we would just uncomment those lines. in that case, the code would need modification again.

Let me know your thoughts.

Reply:
As discussed, made minimal changes to test module code and commented out the EC pool config from test suites
 - [suites/pacific/rados/tier-2_rados_test-pool-functionalities.yaml](https://github.com/red-hat-storage/cephci/pull/2396/files#diff-68a266ae522e5980e08fc149ad3966235339f647742e62621235f0fe4ebc926d)
 - [suites/quincy/rados/tier-2_rados_test-pool-functionalities.yaml](https://github.com/red-hat-storage/cephci/pull/2396/files#diff-c07e04ddcd0a5c7fa0afb624ef57765d7ee4dbb52695be18327692119a197942)